File: sources/common/src/main/java/fr/unice/polytech/si5/al/e/model/Customer.java
package fr.unice.polytech.si5.al.e.model;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.Set;

@Entity
public class Customer {
    @Id
    @GeneratedValue
    private int id;

    @NotNull
    private String name;

    @OneToMany(cascade = CascadeType.PERSIST)
    private Set<Item> items;

    @OneToMany(mappedBy = "customer", cascade = CascadeType.PERSIST)
    private Set<Travel> shipments;

    @OneToMany(mappedBy = "transporter", cascade = CascadeType.PERSIST)
    private Set<Travel> transports;

    public Customer() {
        this.items = new HashSet<>();
        this.shipments = new HashSet<>();
        this.transports = new HashSet<>();
    }

    public void addTravel(Travel travel) {
        shipments.add(travel);
    }

    public void chooseTravel(Travel travel) {
        transports.add(travel);
    }

    public void addItem(Item item) {
        items.add(item);
    }

    public Set<Item> getItems() {
        return items;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public int hashCode() {
        return this.id;
    }

    @Override
    public boolean equals(Object obj) {
        if(obj == this) return true;
        if(!(obj instanceof Customer)) return false;

        return ((Customer) obj).id == this.id;
    }
}

File: blablamoveService/typescript/main/src/entity/Customer.ts
import { Comparable } from "./Comparable";
import { Item } from "./Item";
import { Travel } from "./Travel";

export class Customer implements Comparable {
    private id: string;
    private name: string;
    private email: string;
    private phone: number;
    private items: Item[];
    private shipments: Travel[];
    private transports: Travel[];

    public get $id(): string {
        return this.id;
    }

    public set $id(value: string) {
        this.id = value;
    }

    public get $name(): string {
        return this.name;
    }

    public set $name(value: string) {
        this.name = value;
    }

    public get $email(): string {
        return this.email;
    }

    public set $email(value: string) {
        this.email = value;
    }

    public get $phone(): number {
        return this.phone;
    }

    public set $phone(value: number) {
        this.phone = value;
    }

    constructor() {
        this.items = [];
        this.shipments = [];
        this.transports = [];
    }

    public addTravel(travel: Travel): void {
        this.shipments.push(travel);
    }

    public chooseTravel(travel: Travel): void {
        this.transports.push(travel);
    }

    public addItem(item: Item): void {
        this.items.push(item);
    }

    equal(object: any): boolean {
        if (!(object instanceof Customer)) return false;
        return this.id === object.id;
    }
}
